pkg/apis/networking/v1alpha1/certificate_validation.go

```diff
@@ -44,5 +46,20 @@ func (spec *CertificateSpec) Validate(ctx context.Context) (all *apis.FieldError
 	if spec.SecretName == "" {
 		all = all.Also(apis.ErrMissingField("secretName"))
 	}
+
+	if spec.Domain != "" && len(spec.DNSNames) != 0 {
+		valid := false
+		for _, dnsName := range spec.DNSNames {
+			if strings.HasSuffix(dnsName, fmt.Sprintf(".%s", spec.Domain)) {
```

Comment:
Rather than calling `fmt.Sprintf()` in the loop, do it once outside the loop.

Also, you should be able to use string `+`, rather than needing to pull in the `fmt` module:

```golang
if spec.Domain != "" {  // Already added an error for missing dnsNames
  domainSuffix := "." + spec.Domain
  err := apis.ErrInvalidValue(...)
  for _, dnsName := range spec.DNSNames {
    if strings.HasSuffix(dnsName, domainSuffix) {
      err = nil
      break
    }
  }
  all = all.Also(err)  // nil FieldError is a no-op
}
```

Reply:
ah right, I'll fix that